common/app/model/content.scala

```diff
@@ -55,6 +55,11 @@ class Content protected (val apiContent: ApiContentWithMeta) extends Trail with
   lazy val productionOffice: Option[String] = delegate.safeFields.get("productionOffice")
   lazy val displayHint: String = fields.getOrElse("displayHint", "")
 
+  override lazy val membershipAccess: Option[String] = fields.get("membershipAccess")
+  override lazy val requiresMembershipAccess: Boolean = {
+    conf.Switches.MembersAreaSwitch.isSwitchedOn && membershipAccess.nonEmpty && url.contains("/membership/")
```

Comment:
@gklopper Moved the logic for requiring membership access into a single location and added restriction to only apply to content under `/membership/*` (also used for tests)